### PR TITLE
[RFC] Moved linalg tests into better named files

### PR DESCRIPTION
**Pull request by kshyatt**
Many of the `linalg` tests were in files with opaque names like `linalg1.jl`. I've split up the tests so that all the QR tests are together in `test/linalg/qr.jl`. Some of the tests were being run over and over needlessly because of the complicated structure of `linalg1.jl`. Almost all the changes here are just copying code around to new files.

Some of the sparse linear algebra was tested in these dense linalg files. I've moved that out to where it belongs.

There are a lot of changes in this commit because it was really hard to try to split these mega-files into smaller pieces iteratively. I can go back and try to split this into multiple commits which will independently pass tests if people want.

Running on my 2012 MBA, the old version took 678 seconds to run, and the new version took 746 seconds.

If the decision is made to merge this, we should wait until after coverage is working again so we can make sure I haven't totally destroyed our linalg coverage.

cc: @andreasnoack @jiahao @tkelman @StefanKarpinski

### CODE REVIEW

**Comment by ScottPJones** [comment]
Did you try to make the test names match the corresponding names in base?
I tried to do that with my strings & unicode reorg, and I noticed that @StefanKarpinski also recommended doing something like that.  Your reorg sounds nice!

**Comment by tkelman** [comment]
Did you do the reorganization entirely by hand? For moving around a large amount of code like this it could help with verifying that nothing is being lost if it can be replicated via a script. Though that might be a fair amount more effort than needed. Depends how long this stays open and how many other test changes it needs to take into account in future rebases.

**Comment by kshyatt** [comment]
Yeah, it was by hand.

**Comment by ViralBShah** [comment]
It seems like a lot of work to have to write a script. Is it not good enough if the coverage stays roughly the same? Something like this had to be done eventually.

**Comment by tkelman** [comment]
I think it should be fine if coverage stays the same, and there's a bit of irony to @kshyatt saying "make sure I haven't totally destroyed our linalg coverage," but coverage runs on base are not running correctly right now due to https://github.com/jakebolewski/JuliaParser.jl/issues/19.

**Comment by ScottPJones** [comment]
@tkelman If indeed #11333 is causing this to fail, couldn't it just be removed for now?  It seems that being able to run coverage while trying to get ready for a release candidate is more important than allowing some new syntax.

**Comment by tkelman** [comment]
Coverage didn't start failing when the PR that enabled the syntax was first merged in May, coverage started failing about a week ago when that syntax first started getting used in base. So it would be a much better idea to just find a workaround alternate syntax at the location it's getting used, rather than revert parser changes.

**Comment by yuyichao** [comment]
@tkelman A few days ago @jakebolewski said he is working on it. Not sure about the status now.

**Comment by StefanKarpinski** [comment]
Or just fix JuliaParser.

+1 to this reorg

**Comment by tkelman** [comment]
> we should wait until after coverage is working again

I guess @kshyatt should have titled this WIP if she didn't want it merged yet? Probably fine.

**Comment by kshyatt** [comment]
I don't mind it being merged. If coverage goes down we can fix it. I think it will be much easier to diagnose coverage issues now.

**Comment by andreasnoack** [comment]
Sorry about that. I missed the last part. I don't think it is worth reverting, but we should probably run compare coverage to the parent when JuliaParse is working again.

**Comment by ScottPJones** [comment]
Any word from @jakebolewski on status?  I'm tempted to look at it, if he's on vacation or something.

File: test/linalg/special.jl [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#This file is a part of Julia. License is MIT: http://julialang.org/license
```
Comment (tkelman) [inline test/linalg/special.jl]: This needs to start with `# This file`, the space between `#` and `This` is significant to the `add_license_to_files.jl` script.

**Pull request merged**